File: kits/kotlin/simple/bot/src/main/kotlin/luxaibot/lux/CityTile.kt
package luxaibot.lux

class CityTile(var team: Int, var cityid: String, x: Int, y: Int, cooldown: Double) {
    var pos: Position
    var cooldown: Double
    fun canAct(): Boolean {
        return cooldown < 1
    }

    fun research(): String {
        return java.lang.String.format("r %d %d", pos.x, pos.y)
    }

    fun buildWorker(): String {
        return java.lang.String.format("bw %d %d", pos.x, pos.y)
    }

    fun buildCart(): String {
        return java.lang.String.format("bc %d %d", pos.x, pos.y)
    }

    init {
        pos = Position(x, y)
        this.cooldown = cooldown
    }
}

File: kits/csharp/simple/Bot/Lux/CityTile.cs
namespace Bot.Lux
{
	public class CityTile
	{
		public string CityId;
		public int Team;
		public Position Pos;
		public double Cooldown;
		public CityTile(int teamId, string cityId, int x, int y, double cooldown)
		{
			CityId = cityId;
			Team = teamId;
			Pos = new Position(x, y);
			Cooldown = cooldown;
		}

		public bool CanAct()
		{
			return Cooldown < 1;
		}

		public string Research()
		{
			return $"r {Pos.X:d} {Pos.Y:d}";
		}

		public string BuildWorker()
		{
			return $"bw {Pos.X:d} {Pos.Y:d}";
		}

		public string BuildCart()
		{
			return $"bc {Pos.X:d} {Pos.Y:d}";
		}
	}
}
